client/my-sites/pages/page/index.js

```diff
@@ -342,10 +344,61 @@ class Page extends Component {
 		}
 
 		return (
-			<PopoverMenuItem onClick={ this.exportPage }>
-				<Gridicon icon="cloud-download" size={ 18 } />
-				{ this.props.translate( 'Export page' ) }
-			</PopoverMenuItem>
+			<>
+				<MenuSeparator key="separator" />
+				<PopoverMenuItem onClick={ this.exportPage }>
+					<Gridicon icon="cloud-download" size={ 18 } />
+					{ this.props.translate( 'Export page' ) }
+				</PopoverMenuItem>
+			</>
+		);
+	}
+
+	setComingSoonPage = () => {
+		const pageId = this.props.isComingSoonPage ? 0 : this.props.page.ID;
+
+		this.props.updateSiteFrontPage( this.props.siteId, {
+			wpcom_public_coming_soon_page_id: pageId,
+		} );
+
+		// records whether a page has been set as 'coming soon'
+		// and the paid plan slug
+		this.props.recordTracksEvent( 'calypso_coming_soon_set_page', {
```

Comment:
todo: this will have to be registered https://fieldguide.automattic.com/tracks/tracks-for-developers-product-teams/registering-tracks-events/